src/extensions/posthog.ts

```diff
@@ -1,79 +1,81 @@
+    distinct_id: string
+    event: string
+    timestamp: string
+    properties: Properties
+    team_id: number
+    uuid: string
+}
+
 export interface DummyPostHog {
     capture(event: string, properties?: Record<string, any>): void
 }
 
 export function createPosthog(server: PluginsServer, pluginConfig: PluginConfig): DummyPostHog {
     const distinctId = pluginConfig.plugin?.name || `plugin-id-${pluginConfig.plugin_id}`
 
-    const client = server.KAFKA_ENABLED ? null : new Client(server.redis, server.PLUGINS_CELERY_QUEUE) // Redis
-    const producer = server.KAFKA_ENABLED ? server.kafka!.producer() : null // Kafka
-    producer?.connect()
+    let sendEvent: (data: InternalData) => Promise<void>
 
-    function sendEventRedis(event: string, properties: Record<string, any>, timestamp: string) {
-        const data = {
-            distinct_id: distinctId,
-            event,
-            timestamp,
-            properties: {
-                $lib: 'posthog-plugin-server',
-                $lib_version: version,
-                ...properties,
-            },
+    if (server.KAFKA_ENABLED) {
+        // Sending event to our Kafka>ClickHouse pipeline
+        const producer = server.kafka!.producer()
+        sendEvent = async (data) => {
+            await producer.connect()
+            producer!.send({
+                topic: KAFKA_EVENTS_INGESTION_HANDOFF,
+                messages: [
+                    {
+                        key: data.uuid,
+                        value: JSON.stringify({
+                            distinct_id: data.distinct_id,
+                            ip: '',
+                            site_url: '',
+                            data: JSON.stringify(data),
+                            team_id: pluginConfig.team_id,
+                            now: data.timestamp,
+                            sent_at: data.timestamp,
+                            uuid: data.uuid,
+                        } as RawEventMessage),
+                    },
+                ],
+            })
         }
-
-        client!.sendTask(
-            'posthog.tasks.process_event.process_event_with_plugins',
-            [distinctId, null, null, data, pluginConfig.team_id, timestamp, timestamp],
-            {}
-        )
-    }
-
-    function sendEventKafka(event: string, properties: Record<string, any>, timestamp: string) {
-        const uuid = new UUIDT().toString()
-        const data = {
-            distinct_id: distinctId,
-            event,
-            timestamp,
-            properties: {
-                $lib: 'posthog-plugin-server',
-                $lib_version: version,
-                ...properties,
-            },
+    } else {
+        // Sending event to our Redis>Postgres pipeline
+        const client = new Client(server.redis, server.PLUGINS_CELERY_QUEUE)
+        sendEvent = async (data) => {
+            client.sendTask(
+                'posthog.tasks.process_event.process_event_with_plugins',
```

Comment:
Thinking out loud (and this can be done later), but this string should probably also end up in a `setting`. It's used in 2 places already.